src/PolkitDialog.vala

```diff
@@ -164,6 +164,17 @@ namespace Ag.Widgets {
                 return;
             }
 
+            const string DESKTOP_SCHEMA = "io.elementary.desktop";
+            const string DARK_KEY = "prefer-dark";
+
+            var lookup = SettingsSchemaSource.get_default ().lookup (DESKTOP_SCHEMA, false);
+
+            if (lookup != null) {
+                var desktop_settings = new Settings (DESKTOP_SCHEMA);
+                var gtk_settings = Gtk.Settings.get_default ();
+                desktop_settings.bind (DARK_KEY, gtk_settings, "gtk_application_prefer_dark_theme", SettingsBindFlags.DEFAULT);
```

Comment:
We prefer having properties have `-` instead of `_` in their names, so `gtk-application-prefer-dark-theme`.

Reply:
@donadigo I thought that mapping was done automatically? I'm not creating this property, it's the built-in GTK one.

Reply:
@cassidyjames it's done automatically. What I mean is that GObject uses a naming convention where a property in `Gtk.Settings` of name `gtk_application_prefer_dark_theme` would map to `gtk-application-prefer-dark-theme` when connecting / binding to it.

Reply:
@donadigo Ah, I gotcha now. I didn't realize that. I've pushed that plus an update to use a different key after a lot of discussion and thought. :smile: